cantabular/metadata.go

```diff
@@ -0,0 +1,257 @@
+// to be used against the ext-api.
+
+// These are updated against v1.2 of the metadata schema and would need changing
+// for future releases.
+
+type MetadataQueryResult struct {
+	TableQueryResult   *MetadataTableQuery   `json:"table_query_result"`
+	DatasetQueryResult *MetadataDatasetQuery `json:"dataset_query_result"`
+}
+
+type MetadataTableQuery struct {
+	Service struct {
+		Tables []struct {
+			Name        graphql.String   `json:"name"`
+			DatasetName graphql.String   `json:"dataset_name"`
+			Label       graphql.String   `json:"label"`
+			Description graphql.String   `json:"description"`
+			Vars        []graphql.String `json:"vars"`
+			Meta        struct {
+				Alternate_Geographic_Variables []graphql.String `graphql:"Alternate_Geographic_Variables" json:"alternate_geographic_variables"`
+				Contact                        struct {
+					ContactName    graphql.String `graphql:"Contact_Name" json:"contact_name"`
+					ContactEmail   graphql.String `graphql:"Contact_Email" json:"contact_email"`
+					ContactPhone   graphql.String `graphql:"Contact_Phone" json:"contact_phone"`
+					ContactWebsite graphql.String `graphql:"Contact_Website" json:"contact_website" `
+				} `graphql:"Contact" json:"contact"`
+
+				CensusReleases []struct {
+					CensusReleaseDescription graphql.String `graphql:"Census_Release_Description" json:"census_release_description" `
+					CensusReleaseNumber      graphql.String `graphql:"Census_Release_Number" json:"census_release_number" `
+					ReleaseDate              graphql.String `graphql:"Release_Date" json:"release_date" `
+				} `graphql:"Census_Releases" json:"census_releases"`
+
+				DatasetMnemonic2011 graphql.String `graphql:"Dataset_Mnemonic_2011" json:"dataset_mnemonic2011" `
+				DatasetPopulation   graphql.String `graphql:"Dataset_Population" json:"dataset_population"`
+				GeographicCoverage  graphql.String `graphql:"Geographic_Coverage" json:"geographic_coverage"`
+				LastUpdated         graphql.String `graphql:"Last_Updated" json:"last_updated"`
+				Observation_Type    struct {
+					Observation_Type_Description graphql.String `graphql:"Observation_Type_Description" json:"observation_type_description"`
+
+					Observation_Type_Label graphql.String `graphql:"Observation_Type_Label" json:"observation_type_label"`
+
+					Decimal_Places graphql.String `graphql:"Decimal_Places" json:"decimal_places"`
+
+					Prefix graphql.String `graphql:"Prefix" json:"prefix"`
+
+					Suffix graphql.String `graphql:"Suffix" json:"suffix"`
+
+					FillTrailingSpaces graphql.String `graphql:"FillTrailingSpaces" json:"fill_trailing_spaces"`
+
+					NegativeSign graphql.String `graphql:"NegativeSign" json:"negative_sign"`
+
+					Observation_Type_Code graphql.String `graphql:"Observation_Type_Code" json:"observation_type_code"`
+				} `graphql:"Observation_Type" json:"observation_type"`
+
+				//Keywords            []graphql.String `graphql:"Keywords" json:"keywords"`
```

Comment:
Should we remove the commented out code?

Reply:
done